app/src/main/kotlin/PushYourLuck/PushYourLuckMDP.kt

```diff
@@ -0,0 +1,37 @@
+package PushYourLuck
+
+import MDP
+
+class PushYourLuckMDP(val nDice: Int, val nSides: Int): MDP<PushYourLuckState, PushYourLuckAction>(){
+
+    var diceObject = DiceClass(nDice = nDice, nSides = nSides)
+    var currentReward = 0.0
+
+    override fun initialState(): PushYourLuckState {
+        return PushYourLuckState(diceObject.markedSides)
+    }
+
+    override fun transition(state: PushYourLuckState, action: PushYourLuckAction) : PushYourLuckState {
+
+        if (action == PushYourLuckAction.ROLL){
```

Comment:
Use `when` which is a switch equivalent. That way it enumerates all possible actions and you don't need to throw the exception for invalid input.